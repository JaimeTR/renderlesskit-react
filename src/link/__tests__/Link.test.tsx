import * as React from "react";
import { axe, render } from "reakit-test-utils";

import { Link } from "../index";

describe("ReakitLink", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Link href="#">link</Link>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly with isExternal", () => {
    const { asFragment } = render(
      <Link href="#" isExternal>
        link
      </Link>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly with isExternal & disabled", () => {
    const { asFragment } = render(
      <Link href="#" isExternal disabled>
        link
      </Link>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("should render correctly with non native link tag", () => {
    const { asFragment } = render(
      <Link as="span" onClick={() => alert("Custom Link")}>
        Reakit
      </Link>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Link renders with no a11y violations", async () => {
    const { container } = render(<Link href="#">link</Link>);
    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });
});
