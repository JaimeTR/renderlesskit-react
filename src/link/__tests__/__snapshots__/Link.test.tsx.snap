// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReakitLink should render correctly 1`] = `
<DocumentFragment>
  <a
    href="#"
  >
    link
  </a>
</DocumentFragment>
`;

exports[`ReakitLink should render correctly with isExternal & disabled 1`] = `
<DocumentFragment>
  <a
    aria-disabled="true"
    href="#"
    rel="noopener noreferrer"
    style="pointer-events: none;"
    tabindex="-1"
    target="_blank"
  >
    link
  </a>
</DocumentFragment>
`;

exports[`ReakitLink should render correctly with isExternal 1`] = `
<DocumentFragment>
  <a
    href="#"
    rel="noopener noreferrer"
    target="_blank"
  >
    link
  </a>
</DocumentFragment>
`;

exports[`ReakitLink should render correctly with non native link tag 1`] = `
<DocumentFragment>
  <span
    role="link"
    tabindex="0"
  >
    Reakit
  </span>
</DocumentFragment>
`;
